addons/web/static/src/legacy/js/views/kanban/kanban_record.js

```diff
@@ -610,31 +610,35 @@ var KanbanRecord = Widget.extend(WidgetAdapterMixin, {
      * @returns {Object} transformed record data
      */
     _transformRecord: function (recordData) {
-        var self = this;
-        var new_record = {};
-        _.each(this.state.getFieldNames(), function (name) {
-            var value = recordData[name];
-            var r = _.clone(self.fields[name] || {});
+        const new_record = Object.create(null);
+        for (const name of this.state.getFieldNames()) {
+            const value = recordData[name];
+            const r = new_record[name] = _.clone(this.fields[name] || {});
 
-            if ((r.type === 'date' || r.type === 'datetime') && value) {
-                r.raw_value = value.toDate();
-            } else if (r.type === 'one2many' || r.type === 'many2many') {
+            r.raw_value = value;
+            switch (r.type) {
+            case 'date': case 'datetime':
+                r.raw_value = value && value.toDate();
+                break;
+            case 'one2many': case 'many2many':
                 r.raw_value = value.count ? value.res_ids : [];
-            } else if (r.type === 'many2one') {
+                break;
+            case 'many2one':
                 r.raw_value = value && value.res_id || false;
-            } else {
-                r.raw_value = value;
+                break;
             }
 
-            if (r.type) {
-                var formatter = field_utils.format[r.type];
-                r.value = formatter(value, self.fields[name], recordData, self.state);
-            } else {
+            switch (r.type) {
```

Comment:
why two `switch`?

Reply:
Because I just converted the old conditionals to switches, but you're right that they should probably be mergeable.

Reply:
Looking at the code again, I think I didn't merge it because the two switches don't really overlap, so all the types which have a special case for `raw_value` would need to duplicate the generic case for `value`.